screenwriter/screenwriter.go

```diff
@@ -0,0 +1,60 @@
+package screenwriter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type ScreenWriter struct {
+	writer   io.StringWriter
+	previous string
+}
+
+func New(writer io.StringWriter) *ScreenWriter {
+	return &ScreenWriter{
+		writer: writer,
+	}
+}
+
+func (w *ScreenWriter) ClearPrevious() error {
+	if w.previous == "" {
+		return nil
+	}
+
+	// Get number of lines in previous
+	lines := strings.Count(w.previous, "\n")
+
+	// Handle the final line
+	finalLineLength := 0
```

Comment:
This seems unnecessary to me: we reassign it before using it, so could we declare it later (or just inline the expression where we need it).